PollingAwaitStrategy throws NPE when no portBinding is provided

To my understanding you don't need to explicitly bind ports to reach a container inside docker it self, only if you need to reach it from the outside. 

Since the polling strategy only needs the internal ports to operate anyway, it should inspect the InspectContainerResponse.ContainerConfig.ExposedPorts, not HostConfig to find the ports exposed. 

```java
Caused by: java.lang.NullPointerException
	at org.arquillian.cube.docker.impl.util.BindingUtil.binding(BindingUtil.java:31)
	at org.arquillian.cube.docker.impl.model.DockerCube.bindings(DockerCube.java:143)
	at org.arquillian.cube.docker.impl.await.PollingAwaitStrategy.await(PollingAwaitStrategy.java:91)
	at org.arquillian.cube.docker.impl.model.DockerCube.start(DockerCube.java:93)
	... 18 more
```

--- a/docker/docker/src/main/java/org/arquillian/cube/docker/impl/await/PollingAwaitStrategy.java
+++ b/docker/docker/src/main/java/org/arquillian/cube/docker/impl/await/PollingAwaitStrategy.java
@@ -95,6 +95,9 @@ public class PollingAwaitStrategy implements AwaitStrategy {
 
             switch(this.type) {
                 case "ping": {
+                    if(ports.getBindingPort() == -1) {
+                        throw new IllegalArgumentException("Can not use polling of type " + type + " on non externally bound port " + ports.getExposedPort());
+                    }
                     if (!Ping.ping(bindings.getIP(), ports.getBindingPort(), this.pollIterations, this.sleepPollTime,
                             this.timeUnit)) {
                         return false;

--- a/docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/containerobject/CubeContainerObjectTestEnricher.java
+++ b/docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/containerobject/CubeContainerObjectTestEnricher.java
@@ -178,7 +178,7 @@ public class CubeContainerObjectTestEnricher implements TestEnricher {
             if (hostPortValue > 0) {
                 final Binding.PortBinding bindingForExposedPort = cube.bindings().getBindingForExposedPort(hostPortValue);
 
-                if (bindingForExposedPort != null) {
+                if (bindingForExposedPort != null && bindingForExposedPort.getBindingPort() != -1) {
                     field.set(containerObjectInstance, bindingForExposedPort.getBindingPort());
                 } else {
                     throw new IllegalArgumentException(String.format("Container Object %s contains field %s annotated with %s but exposed port %s is not exposed on container object.", containerObjectInstance.getClass().getSimpleName(), field.getName(), HostPort.class.getSimpleName(), hostPortValue));

--- a/docker/docker/src/main/java/org/arquillian/cube/docker/impl/util/BindingUtil.java
+++ b/docker/docker/src/main/java/org/arquillian/cube/docker/impl/util/BindingUtil.java
@@ -8,6 +8,7 @@ import org.arquillian.cube.docker.impl.docker.DockerClientExecutor;
 import org.arquillian.cube.spi.Binding;
 
 import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.github.dockerjava.api.model.ContainerConfig;
 import com.github.dockerjava.api.model.ExposedPort;
 import com.github.dockerjava.api.model.HostConfig;
 
@@ -22,16 +23,22 @@ public final class BindingUtil {
     public static Binding binding(DockerClientExecutor executor, String cubeId) {
         InspectContainerResponse inspectResponse = executor.getDockerClient().inspectContainerCmd( cubeId ).exec();
 
-        HostConfig hostConfig = inspectResponse.getHostConfig();
-
         String dockerIp = getDockerServerIp(executor);
         Binding binding = new Binding(dockerIp);
 
-        for (Entry<ExposedPort, com.github.dockerjava.api.model.Ports.Binding[]> bind : hostConfig.getPortBindings()
-                .getBindings().entrySet()) {
-            com.github.dockerjava.api.model.Ports.Binding[] allBindings = bind.getValue();
-            for (com.github.dockerjava.api.model.Ports.Binding bindings : allBindings) {
-                binding.addPortBinding(bind.getKey().getPort(), bindings.getHostPort());
+        HostConfig hostConfig = inspectResponse.getHostConfig();
+        if(hostConfig.getPortBindings() != null) {
+            for (Entry<ExposedPort, com.github.dockerjava.api.model.Ports.Binding[]> bind : hostConfig.getPortBindings()
+                    .getBindings().entrySet()) {
+                com.github.dockerjava.api.model.Ports.Binding[] allBindings = bind.getValue();
+                for (com.github.dockerjava.api.model.Ports.Binding bindings : allBindings) {
+                    binding.addPortBinding(bind.getKey().getPort(), bindings.getHostPort());
+                }
+            }
+        } else {
+            ContainerConfig connectionConfig = inspectResponse.getConfig();
+            for(ExposedPort port : connectionConfig.getExposedPorts()) {
+                binding.addPortBinding(port.getPort(), -1);
             }
         }
         return binding;

